third_party/blink/renderer/modules/input_control/InputControl.java

```diff
@@ -58,7 +70,50 @@ private static InputControl CreateInputControl() {
     }
 
     @CalledByNative
-    public void SendInput(String type, int x, int y, int code) {
+    public void SendMouseInput(String type, int x, int y) {
+        EnterpriseDeviceManager edm = EnterpriseDeviceManager.getInstance(
+          ContextUtils.getApplicationContext());
+        RemoteInjection remoteInjection = edm.getRemoteInjection();
+        int action = -1;
+
+        if(type.equals("mousedown"))
+            action = MotionEvent.ACTION_DOWN;
+        else if(type.equals("mousemove"))
+            action = MotionEvent.ACTION_MOVE;
+        else if(type.equals("mouseup"))
+            action = MotionEvent.ACTION_UP;
+
+        InjectMouseEvent(remoteInjection, type, action, x, y);
+    }
+
+    @CalledByNative
+    public void SendTouchInput(String type, String json) {
+        EnterpriseDeviceManager edm = EnterpriseDeviceManager.getInstance(
+          ContextUtils.getApplicationContext());
+        RemoteInjection remoteInjection = edm.getRemoteInjection();
+        int action = -1;
+
+        if (type.equals("touchstart") && mNumTouch < 1) {
```

Comment:
How about this?
if (type.equals("touchstart")) {
  if (mNumTouch < 1) {
    ....
  } else {
    .....
  }
} else if (type.equals("touchend")) {
  ....
} else if (type.equals("touchmove")) {
  ....
}

Reply:
Thank you 👍 applied